sana/mrs/json.py

```diff
@@ -1189,5 +1191,151 @@ def syc_encounters(request, patient_id, encounters=None):
+                'data': cjson.decode(notification.to_json()),
+        }
+    except ObjectDoesNotExist as e:
+        et, val, tb = sys.exc_info()
+        trace = traceback.format_tb(tb)
+        error = "Exception : %s %s %s" % (et, val, trace[0])
+        for tbm in trace:
+            logging.error(tbm)
+        logging.error("Got exception while fetching notification: %s" % e)
+        response = {'status': 'FAILURE',
+            'data': "Problem while getting notification: %s" % e,
+        }
+    except MultipleObjectsReturned:
+        try:
+            notification = Notification.objects.filter(procedure_id=id)
+            logging.info("we finished getting the notification")
+            response = {'status': 'SUCCESS',
+                    'data': [cjson.decode(d.to_json()) for d in notification],
+            }
+        except Exception as e:
+            et, val, tb = sys.exc_info()
+            trace = traceback.format_tb(tb)
+            error = "Exception : %s %s %s" % (et, val, trace[0])
+            for tbm in trace:
+                logging.error(tbm)
+            logging.error("Got exception while fetching notification: %s" % e)
+            response = {'status': 'FAILURE',
+                'data': "Problem while getting notification: %s" % e,
+            }
+
+    except Exception, e:
+        et, val, tb = sys.exc_info()
+        trace = traceback.format_tb(tb)
+        error = "Exception : %s %s %s" % (et, val, trace[0])
+        for tbm in trace:
+            logging.error(tbm)
+        logging.error("Got exception while fetching notification: %s" % e)
+        response = {'status': 'FAILURE',
+            'data': "Problem while getting notification: %s" % e,
+        }
+    return HttpResponse(cjson.encode(response), mimetype=("application/json; charset=utf-8"))
+
+
+
+@enable_logging
+def notification_list(request):
+    """For synching notifications with mobile clients.
 
+    Request Params
+        username
+            A valid username.
+        password
+            A valid password.
 
+    Parameters:
+        request
+            A client request for patient list
+    """
+    logging.info("entering notification list proc")
+    try:
+        data = Notification.objects.all()
+        logging.info("we finished getting the notification list")
+        response = {'status': 'SUCCESS',
+            'data': [cjson.decode(d.to_json()) for d in data],
+        }
+    except Exception, e:
+        et, val, tb = sys.exc_info()
+        trace = traceback.format_tb(tb)
+        error = "Exception : %s %s %s" % (et, val, trace[0])
+        for tbm in trace:
+            logging.error(tbm)
+        logging.error("Got exception while fetching notification list: %s" % e)
+        response = {
+            'status': 'FAILURE',
+            'data': "Problem while getting notification list: %s" % e,
+        }
+    return HttpResponse(cjson.encode(response), mimetype=("application/json; charset=utf-8"))
```

Comment:
Nothing to do about this while we're on Django 1.3, but worth noting that Django 1.5 [deprecates the `mimetype` argument in the `HttpResponse` constructor](https://docs.djangoproject.com/en/1.5/ref/request-response/#id5) in favor of `content_type` and Django 1.6 removes it entirely. Will need to track all these down and replace then when/if we upgrade Djangos.